package com.diffblue.deeptestutils;

import org.junit.Assert;
import org.junit.rules.ExpectedException;
import org.junit.Test;

import java.lang.reflect.Array;
import java.lang.reflect.Constructor;
import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

import com.diffblue.deeptestutils.CompareWithFieldList;
import com.diffblue.deeptestutils.Reflector;

public class UnexpectedValueExceptionTest {

  @org.junit.Rule
  public ExpectedException thrown = ExpectedException.none();

  /* testedClasses: UnexpectedValueException */


  /*
   * Test generated by Diffblue Deeptest.
   * This test case covers the entire function.
   */

  @org.junit.Test
  public void com_diffblue_deeptestutils_UnexpectedValueException__init__000_ffe27027ac561668() throws Throwable {

    Object instance;
    {
      /* Arrange */
      String errorMsg = null;

      /* Act */
      /* Creating object to test constructor */
      Class<?> c = Reflector.forName("com.diffblue.deeptestutils.UnexpectedValueException");
      Constructor<?> ctor = c.getDeclaredConstructor(Reflector.forName("java.lang.String"));
      ctor.setAccessible(true);
      instance = ctor.newInstance(errorMsg);

      /* Method return type is void, or not expected to return */
    }
  }
}
